### PR TITLE
feat(fix): related things

### PR DESCRIPTION
**Pull request by IRacle1**
## Description
**Describe the changes** 
- fixed `PumpBarrelMagazine::IsCocked`
- fixed `NormalMagazine::Ammo` setter (dont set when magazine is not inserted)
image
- fixed microhid bug by creating in `Item.Create`
- fixed rooms generation
- fixed logic for `AmmoDrain == 0`

**What is the current behavior?** (You can also link to an open issue here)
😔

**What is the new behavior?** (if this is a feature change)
🥹

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no🥴

**Other information**:

<br />

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [x] I have checked no IL patching errors in the console

### Other
- [x] Still requires more testing

### CODE REVIEW

**Comment by LumiFae** [comment]
what the hell is a feat(fix)

**Comment by IRacle1** [comment]
> what the hell is a feat(fix)

feat and fix🤨

**Comment by LumiFae** [comment]
> > what the hell is a feat(fix)
> 
> feat and fix🤨

yeah that's not how conventional commits is... if it has a feature (and fixes) then it's just feat, if it has just fixes, fix. also the brackets stuff is for the scope, so it will be smth like feat(API) for api features, feat(Events) for event features :sob:

**Comment by LumiFae** [comment]
you'll get used to it the more you do it, but yes i think the way conventional commits is used here is wrong, but can also be useful, the reason it's called conventional commits is because it's supposed to be used for commit titles not PR titles but last time I raised that everyone went wild so I'm not doing that again. i can help with a title when this pr is no longer a draft, just ping me. unless it's mostly finished now then I can figure out smth now

**Review by Misfiy — APPROVED** [review]

File: EXILED/Exiled.Events/Patches/Generic/AmmoDrain.cs [hunk 1]
```diff
@@ -224,17 +241,17 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
                     new(OpCodes.Ldloc_S, ammoDrain.LocalIndex),
-                    new(OpCodes.Mul),
+                    new(OpCodes.Call, Method(typeof(AmmoDrainAutomatic), nameof(AmmoDrainAutomatic.GetAmmoDrainLimit))),
                 });
 
             offset = 0;
-            index = newInstructions.FindIndex(i => i.Calls(Method(typeof(IPrimaryAmmoContainerModule), nameof(IPrimaryAmmoContainerModule.ServerModifyAmmo)))) + offset;
+            index = newInstructions.FindIndex(i => i.Calls(Method(typeof(InventoryExtensions), nameof(InventoryExtensions.ServerAddAmmo)))) + offset;
 
             // and divide ammo that are inserting in magazine, to implement AmmoDrain
```
Comment (louis1706) [inline EXILED/Exiled.Events/Patches/Generic/AmmoDrain.cs:250]:
```suggestion
            // and multiply ammo that are inserting in magazine, to implement AmmoDrain
```

**Comment by VALERA771** [comment]
@IRacle1 is it ready for merging?

**Comment by IRacle1** [comment]
Yep

**Review by VALERA771 — APPROVED** [review]

**Pull request merged**